src/PrinterLog.php

```diff
@@ -108,14 +109,20 @@ public static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $
      *
      * @return array
      */
-    public function getMetrics(Printer $printer, $user_filters = []): array
+    public function getMetrics(Printer $printer, $user_filters = [], $interval = 'P1Y', $start_date = null, $end_date = new DateTime(), $format = 'dynamic'): array
     {
         global $DB;
 
-        $bdate = new DateTime();
-        $bdate->sub(new DateInterval('P1Y'));
+        if (!$start_date) {
+            $start_date = new DateTime();
+            $start_date->sub(new DateInterval($interval));
+        }
+
         $filters = [
-            'date' => ['>', $bdate->format('Y-m-d')]
+            'date' => ['>=', $start_date->format('Y-m-d')],
+            'AND' => [
+                'date' => ['<=', $end_date->format('Y-m-d')]
+            ],
```

Comment:
If you are looking to avoid duplicated key names, you can simply wrap it into another array:

```suggestion
           ['date' => ['>=', $start_date->format('Y-m-d')]],
           ['date' => ['<=', $end_date->format('Y-m-d')]]
```